parachain/primitives/core/src/inbound.rs

```diff
@@ -32,12 +32,36 @@ pub type MessageNonce = u64;
 /// A bridge message from the Gateway contract on Ethereum
 #[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
 pub struct Message {
-	/// RLP-encoded event log
-	pub data: Vec<u8>,
+	/// Event log emitted by Gateway contract
+	pub event_log: Log,
 	/// Inclusion proof for a transaction receipt containing the event log
 	pub proof: Proof,
 }
 
+const MAX_TOPICS: usize = 4;
+
+#[derive(Clone, RuntimeDebug)]
+pub enum LogValidationError {
+	TooManyTopics,
+}
+
+/// Event log
+#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct Log {
+	pub address: H160,
+	pub topics: Vec<H256>,
+	pub data: Vec<u8>,
+}
```

Comment:
Also we have duplicated definition for the `Log` structure in 

https://github.com/Snowfork/snowbridge/blob/6b592235b907e00e9fe6ab4446ff48564bd41474/parachain/primitives/ethereum/src/log.rs#L8-L13

So maybe we can make some furthur refactoring.